### ISSUE TITLE
`--rq` option to rebuild from branch to leaf

### ISSUE THREAD

**Issue by LecrisUT** [juhp/fbrnch#50]
Just to have a tracker for the discussion on matrix. The feature request is to be able to `bump`, `push`, `build`, etc. starting from a branch update, e.g. a rust library, up to the leaf consumers (rust binary in the example). Would also be useful to create a file or outputting to text so that we can start restart from different points in `bump`, `push`, `build` step without having to recompile the list each time.

Another idea in the discussion was that tools like `fedrq wr` cannot be 100% accurate due to arch specific dependencies, but this could be complemented by scraping a `build.log`/`root.log`. Maybe would require to store some database of dependencies locally, though no complaints if there was a hosted one or directly inside `koji` ;)

**Comment by juhp** [comment]
> The feature request is to be able to `bump`, `push`, `build`, etc. starting from a branch update, e.g. a rust library, up to the leaf consumers (rust binary in the example). Would also be useful to create a file or outputting to text so that we can start restart from different points in `bump`, `push`, `build` step without having to recompile the list each time.

That was a little hard to understand - re-reading, okay I think you talking about a dependency tree of packages.
But I am still not completely clear what you are after here: sounds almost like you want fbrnch scripting...

Maybe you could give a small concrete real example to make it clear: currently fbrnch only run one command at a time, though there is `fbrnch command ...`.

> Another idea in the discussion was that tools like `fedrq wr` cannot be 100% accurate due to arch specific dependencies, but this could be complemented by scraping a `build.log`/`root.log`. Maybe would require to store some database of dependencies locally, though no complaints if there was a hosted one or directly inside `koji` ;)

Interesting idea - probably good to start from some example again. The thing is once you have built the package
then you can directly query it for dependencies though.

**Comment by LecrisUT** [comment]
Basically, let's say we have LibA that is having a SOVERSION bump, the workflow there is to bump all dependent packages as if queried by `fedrq wr LibA`. 2nd more complex level, let's say we have a rust package update, we need to do the same steps recursively up to bin packages.

**Comment by juhp** [comment]
Ah gotcha: so you want something like `rebuild-dependents` - yes that definitely makes sense and should be doable for trivial cases (if there are major dependency changes of course it gets more complicated, but we can ignore that for now).

**Comment by juhp** [comment]
(Also just for reference there is `fbrnch install --recursive` again based on adjacent package repo checkouts.)

**Comment by juhp** [comment]
(Also `push`ing is already done automatically by `build` and `parallel`.)

**Comment by LecrisUT** [comment]
> (Also `push`ing is already done automatically by `build` and `parallel`.)

Yeah, but we might need to have these supported separately if we don't have write access to all dependencies. Or maybe we can assume #51 would take care of that.
